// ecomme_package/eccommerce.js

exports.libTime = require("timestamp_package");
//this per richiamarlo da fuori this.products
exports.products = products =[{id:1, product:"IphoneX", amount:10},
              {id:2, product:"Samsung8", amount:10},
              {id:3, product:"LgG5", amount:10},
              {id:4, product:"AsusZenfone3", amount:10}];
var countPrd=products.length;
 let buyTime =[]; exports.delete={};
exports.post ={}; exports.put={};

exports.insert = function(product,amount){
        countPrd++;
        products.push({
                  id:countPrd,
                  product:product,
                  amount:amount});
            post={message:"Post is aquired!"}
          return post;
  }

exports.delet = function(id){
      products.splice((id-1),1) //elimina il posto alla pozione 3 e shifta il resto sopra di uno;
      delet={message:"Delete is completed"}
      return delet;
    }
exports.put = function(id, product, amount){
  for(i=0;i<products.length;i++){
    if(id===products[i].id){
      products[i]={
      id:products[i].id,
      product:product,
      amount:amount
      };
      put={message:"The edit is completed"}
    }
  }
  return put;
}

exports.buy = function(id,token){
  for(i=0;i<products.length;i++){
    var count=0;
    if(id===products[i].id){
      if(products[i].amount > 0){
        products[i].amount--;
          count++;
        return buyTime.push({
          id:products[i].id,
          amount:count,
          tokenUser:token,
          buyed:this.libTime.time()
        });
      }
    }
  }
}

  exports.productNow = function() {
    return products;
  }
  exports.productAcquired = function() {
    return buyTime;
  }
